lib/inventory/metadata/metadata.go

```diff
@@ -210,14 +199,35 @@ func (c *fetchConfig) fetchContainerRuntime() string {
 
 // fetchContainerOrchestrator returns kubernetes-${GIT_VERSION} if the instance is
 // running on kubernetes.
+// This function performs the equivalent of the following:
+// curl -k https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT/version | jq .gitVersion
 func (c *fetchConfig) fetchContainerOrchestrator() string {
-	if c.kubeClient == nil {
+	url := fmt.Sprintf("https://%s:%s/version", c.getenv("KUBERNETES_SERVICE_HOST"), c.getenv("KUBERNETES_SERVICE_PORT"))
```

Comment:
Should we check if the env variables are non-empty first?

Do we need to worry about the vars being non-URL-friendly?

Reply:
Force the check to because go's `url.Parse` is not clever and will eventually call weird urls.

https://go.dev/play/p/__dk6CwnvD4